zenoh-flow/src/runtime/resources.rs

```diff
@@ -607,50 +611,50 @@ impl DataStore {
         fid: &str,
         iid: &Uuid,
     ) -> ZFResult<()> {
-        let path = zenoh::Path::try_from(RT_FLOW_PATH!(ROOT_STANDALONE, rtid, fid, iid))?;
-        let ws = self.z.workspace(None).await?;
-        Ok(ws.delete(&path).await?)
+        let path = RT_FLOW_PATH!(ROOT_STANDALONE, rtid, fid, iid);
+
+        Ok(self.z.delete(&path).await?)
     }
 
     pub async fn add_runtime_flow(
         &self,
         rtid: &Uuid,
         flow_instance: &DataFlowRecord,
     ) -> ZFResult<()> {
-        let path = zenoh::Path::try_from(RT_FLOW_PATH!(
+        let path = RT_FLOW_PATH!(
             ROOT_STANDALONE,
             rtid,
             flow_instance.flow,
             flow_instance.uuid
-        ))?;
-        let ws = self.z.workspace(None).await?;
+        );
+
         let encoded_info = serialize_data(flow_instance)?;
-        Ok(ws.put(&path, encoded_info.into()).await?)
+        Ok(self.z.put(&path, encoded_info).await?)
     }
 
     // Registry Related
 
     pub async fn add_graph(&self, graph: &RegistryNode) -> ZFResult<()> {
-        let path = zenoh::Path::try_from(REG_GRAPH_SELECTOR!(ROOT_STANDALONE, &graph.id))?;
-        let ws = self.z.workspace(None).await?;
+        let path = REG_GRAPH_SELECTOR!(ROOT_STANDALONE, &graph.id);
+
         let encoded_info = serialize_data(graph)?;
-        Ok(ws.put(&path, encoded_info.into()).await?)
+        Ok(self.z.put(&path, encoded_info).await?)
     }
 
     pub async fn get_graph(&self, graph_id: &str) -> ZFResult<RegistryNode> {
-        let selector = zenoh::Selector::try_from(REG_GRAPH_SELECTOR!(ROOT_STANDALONE, graph_id))?;
-        let ws = self.z.workspace(None).await?;
-        let mut ds = ws.get(&selector).await?;
+        let selector = REG_GRAPH_SELECTOR!(ROOT_STANDALONE, graph_id);
+
+        let mut ds = self.z.get(&selector).await?;
 
         // Not sure this is needed...
-        let data = ds.collect::<Vec<zenoh::Data>>().await;
+        let data = ds.collect::<Vec<Reply>>().await;
         match data.len() {
             0 => Err(ZFError::Empty),
             _ => {
                 let kv = &data[0];
-                match &kv.value {
-                    zenoh::Value::Raw(_, buf) => {
-                        let ni = deserialize_data::<RegistryNode>(&buf.to_vec())?;
+                match &kv.data.value.encoding {
+                    e if e.starts_with(&Encoding::APP_OCTET_STREAM) => {
+                        let ni = deserialize_data::<RegistryNode>(&kv.data.value.payload.to_vec())?;
                         Ok(ni)
                     }
                     _ => Err(ZFError::DeseralizationError),
```

Comment:
That code is repeated many times, I’m wondering if we could refactor it with a generic type.
For instance:

```rust
async fn get<T>(&self, path: xxx) -> ZFResult<T> {
  let mut ds = self.z.get(&path).await?;
  let data = ds.collect::<Vec<Reply>>().await;
  match data.len() {
    0 => Err(…),
    _ => {
      let kv = &data[0];
      match &kv.data.value.encoding {
        … // use T in this match
      }
    }
  }
}
```

Though I don’t know if we can catch all Zenoh subtleties with one function (`path` vs `selector` for instance, `Vec<T>` vs `T`, etc.).

Reply:
It could be an optimization that can be done right after this PR.
So we can have zenoh flow building correctly ASAP and then we optimize it.